src/wkwebview/mod.rs

```diff
@@ -825,6 +821,96 @@ r#"Object.defineProperty(window, 'ipc', {
     Ok(())
   }
 
+  unsafe fn cookie_from_wkwebview(cookie: &NSHTTPCookie) -> cookie::Cookie<'static> {
+    let name = cookie.name().to_string();
+    let value = cookie.value().to_string();
+
+    let mut cookie_builder = cookie::CookieBuilder::new(name, value);
+
+    let domain = cookie.domain().to_string();
+    cookie_builder = cookie_builder.domain(domain);
+
+    let path = cookie.path().to_string();
+    cookie_builder = cookie_builder.path(path);
+
+    let http_only = cookie.isHTTPOnly();
+    cookie_builder = cookie_builder.http_only(http_only);
+
+    let secure = cookie.isSecure();
+    cookie_builder = cookie_builder.secure(secure);
+
+    let same_site = cookie.sameSitePolicy();
+    let same_site = match same_site {
+      Some(policy) if policy.as_ref() == NSHTTPCookieSameSiteLax => cookie::SameSite::Lax,
+      Some(policy) if policy.as_ref() == NSHTTPCookieSameSiteStrict => cookie::SameSite::Strict,
+      _ => cookie::SameSite::None,
+    };
+    cookie_builder = cookie_builder.same_site(same_site);
+
+    let expires = cookie.expiresDate();
+    let expires = match expires {
+      Some(datetime) => {
+        cookie::time::OffsetDateTime::from_unix_timestamp(datetime.timeIntervalSince1970() as i64)
+          .ok()
+          .map(cookie::Expiration::DateTime)
+      }
+      None => Some(cookie::Expiration::Session),
+    };
+    if let Some(expires) = expires {
+      cookie_builder = cookie_builder.expires(expires);
+    }
+
+    cookie_builder.build()
+  }
+
+  pub fn cookies_for_url(&self, url: &str) -> Result<Vec<cookie::Cookie<'static>>> {
+    let url = url::Url::parse(url)?;
+
+    self.cookies().map(|cookies| {
+      cookies.filter(|cookie: cookie::Cookie| {
+        // domain is the same
+        cookie.domain() == url.domain()
+          // path is the same
+          && cookie.path() == Some(url.path())
+          /// and one of
+          && (
+            // cookie is secure and url is https
+            (cookie.secure() && url.scheme() == "https") ||
+            // or cookie is secure and is localhost
+            (
+              cookie.secure() && url.scheme() == "http" && 
+              (url.domain() == Some("localhost") || url.domain().and_then(|d| Ipv4Addr::from_str(d).ok()).map(|ip| ip.is_loopback()).unwrap_or(false))
+            ) ||
+            // or cookie is not secure
+            (!cookie.secure())
+          )
```

Comment:
not sure if this is too restrictive, tried to check how webkit2gtk cookies_for_url is implementation but I think their implementation is a bit more relaxed ? https://github.com/WebKit/WebKit/blob/c305a86965236fd8052623faaf8804b1424e7f6b/Source/WebKit/UIProcess/API/APIHTTPCookieStore.cpp#L69

Reply:
when in doubt just ask @tweidinger though I think we should just match the behavior from other platforms if possible

Reply:
I think we do match too strict on the path depending on the cookie policy.
For example `/` in the cookie would not match a subpath cookie but it should match any path.

Reply:
Just checking the domain match and checking if secure cookie or not sounds acceptable.

Reply:
alright, pushed a change to match on domain and secure or not. (basically just removed matching the path)